### PR TITLE
Derive or implement Default, PartialOrd, Ord, Hash

### PR DESCRIPTION
**Pull request by dongcarl**

### CODE REVIEW

File: src/sha512.rs [hunk 1]
```diff
@@ -78,6 +78,32 @@ impl PartialEq for Hash {
...
+    fn default() -> Hash {
+        Hash([0; 64])
+    }
+}
+
+use std::cmp::Ordering;
+
+impl PartialOrd for Hash {
+    fn partial_cmp(&self, other: &Hash) -> Option<Ordering> {
+        (&self).partial_cmp(&other)
```
Comment (TheBlueMatt) [inline src/sha512.rs]: wat. I'm confused.

Comment (TheBlueMatt) [reply src/sha512.rs]: Oh, maybe make the slice conversion clear?

Comment (dongcarl) [reply src/sha512.rs]: @TheBlueMatt Eh... Like `(&self[..])`? Kinda defeats the whole purpose of https://github.com/rust-bitcoin/bitcoin_hashes/pull/13

Comment (TheBlueMatt) [reply src/sha512.rs]: I mean in one case its clear, in this case someone (like me) who doesn't know rust as well could *totally* see the compiler just calling partial_cmp recursively directly here...I'd also accept (&self.0).partial_cmp(&other.0).

**Comment by dongcarl** [comment]
Concern addressed and rebased.

**Pull request merged**